packages/keyring-eth-hd/src/index.ts

```diff
@@ -0,0 +1 @@
+export * from './hd-keyring';
```

Comment:
Making this change now makes it difficult to review unless we open it locally and compare. Could you separate this into two steps? First, focus only on the migration, and move the other changes to a second step. This way, we can concentrate solely on the types in this case.

Reply:
I opened this other PR where there is only the migration (without the addition of `index.ts` and the rename to `hd-keyring.ts`): https://github.com/MetaMask/accounts/pull/166 
Unfortunately, as you can see, the file content still differs too much for Github to show it as a rename and edit. So I believe it doesn't bring many advantages to the review process

However, it can also be seen here in this PR at the specific commit where I migrated to TS: https://github.com/MetaMask/accounts/pull/161/commits/ee40095b27647f7b463334eb6d90a3d8614cf484

Though, let me know if you have further suggestions to show a better diff

Reply:
Actually, there is[ this commit](https://github.com/MetaMask/accounts/pull/166/commits/bc89c460ecefb1334378cfc77ce87b8af2e0853a#diff-79137d999831949b43b4c2265b80e69a0f45626d482d2a0ab261241cabb44e68) on the new PR where the changes can be seen much more clearly. I will rebase this PR on top of the other one. Thanks for the suggestion

Reply:
can we use named exports instead of barrel export here as per our recent guidelines https://github.com/MetaMask/core/blob/main/docs/package-guidelines.md#guidelines-for-packages

Reply:
Good idea! done in ae4fa8cd481103c1ae8c2c28f22ee11d1fd66686